### PR TITLE
Chore terms redirection

### PR DESCRIPTION
**Pull request by luchotc**
Depends on https://github.com/mumuki/mumuki-domain/pull/161

This one it's a mix between what we've talked about in the beginning and what we've released last time.

With this one we are only forcing a redirection when user hasn't accepted some role terms. These doesn't include student role term which is considered as a general term, not a role one.
This applies for new role terms or new ones. **This means that every teacher or above or moderator user will be forced to accept the terms, for all organizarions**

When you are redirected, you are forced to accept the terms in order to continue.
We are still accepting all terms when going to the user profile, so there isn't a change in that flow. New users shouldn't ever be redirected as the profile completion is validated before terms redirection.

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Should I have a look at this one too @luchotc ?

**Comment by luchotc** [comment]
Nono, just letting the code here in case we want to implement mandatory acceptance terms! @flbulgarelli

**Comment by luchotc** [comment]
@flbulgarelli Now it's ready to review :smile:

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -15,6 +16,11 @@ def update
     redirect_to user_path, notice: I18n.t(:user_data_updated)
   end
 
+  def accept_profile_terms
+    current_user.accept_profile_terms!
+    redirect_to root_path, notice: I18n.t(:terms_accepted)
```
Comment (luchotc) [inline app/controllers/users_controller.rb:21]: This redirection is not ideal but it'll be fixed by a subsequent PR

Comment (flbulgarelli) [reply app/controllers/users_controller.rb:21]: It is not that bad

File: config/routes.rb [hunk 2]
```diff
@@ -55,6 +55,7 @@
     # All users
     resource :user, only: [:show, :edit]
     get '/user/terms' => 'users#terms'
+    post '/user/terms' => 'users#accept_profile_terms'
```
Comment (flbulgarelli) [inline config/routes.rb:58]: Just to keep things more clear, we could use nested resources like 

```
resource :user, .... do 
   get :terms ....
   ...
end
```

**Review by flbulgarelli — APPROVED** [review]
Looks good. It would be nice to add some capybara tests, don't you thing so?

**Comment by luchotc** [comment]
@julian-berbel could you please take a quick look at this one to validate there's nothing weird?

**Review by julian-berbel — COMMENTED** [review]
Looks great!

**Pull request merged**